daemon/daemon.go

```diff
@@ -100,7 +100,7 @@ func (c *Command) canAccess(r *http.Request, user *auth.UserState) bool {
 			return true
 		}
 
-		if c.PolkitOK != "" {
+		if r.Method != "GET" && c.PolkitOK != "" {
```

Comment:
Hmmm, kind of unsure if all GETs should skip policykit check.

Reply:
Without this change, commands like "snap list" may pop up an interactive dialog requesting authorisation.  The command will still work if the user dismisses the dialog, but we don't want to unnecessarily prompt the user.

Part of the problem here is that there is only a single security policy per resource exported via the API, but in reality we're applying different policies to different request methods sent to that resource.

For example, while `/v2/snaps` entry has the `UserOK` flag set, the real policy is that GET is available to regular users while POST requires admin access.  We want to avoid the polkit check completely for the GET request.  I could probably tighten things up with something like `!(r.Method == "GET" && (c.UserOK || c.GuestOK))`, but that's encoding even more details about the rest of the policy.

Perhaps things would be clearer if the permissions were explicitly specified on a per request method basis?